libbeat/processors/rate_limit/rate_limit.go

```diff
@@ -0,0 +1,133 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package rate_limit
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/jonboulle/clockwork"
+	"github.com/mitchellh/hashstructure"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/logp"
+	"github.com/elastic/beats/v7/libbeat/processors"
+	"github.com/elastic/beats/v7/libbeat/processors/rate_limit/algorithm"
+	jsprocessor "github.com/elastic/beats/v7/libbeat/processors/script/javascript/module/processor"
+)
+
+func init() {
+	processors.RegisterPlugin("rate_limit", New)
+	jsprocessor.RegisterPlugin("RateLimit", New)
+}
+
+const processorName = "rate_limit"
+
+type rateLimit struct {
+	config    Config
+	algorithm algorithm.Algorithm
+	logger    *logp.Logger
+
+	// For testing
+	c clockwork.Clock
```

Comment:
Not used, no need to keep it here.
```suggestion
```

Reply:
Removed in 6e34722eea69bcd18740f97a96b51c39de981cfc.